bofire/strategies/predictives/multi_fidelity.py

```diff
@@ -0,0 +1,70 @@
+import pandas as pd
+
+from bofire.data_models.features.api import TaskInput
+from bofire.data_models.strategies.predictives.multi_fidelity import (
+    MultiFidelityStrategy as DataModel,
+)
+from bofire.strategies.predictives.sobo import SoboStrategy
+
+
+class MultiFidelityStrategy(SoboStrategy):
+    def __init__(self, data_model: DataModel, **kwargs):
+        super().__init__(data_model=data_model, **kwargs)
+        self.task_feature_key = self.domain.inputs.get_keys(TaskInput)[0]
+
+        ft = data_model.fidelity_thresholds
+        M = len(self.domain.inputs.get_by_key(self.task_feature_key).fidelities)  # type: ignore
+        self.fidelity_thresholds = ft if isinstance(ft, list) else [ft] * M
+
+    def _ask(self, candidate_count: int) -> pd.DataFrame:
+        """Generate new candidates (x, m).
+
+        This is a greedy optimization of the acquisition function. We first
+        optimize the acqf for the target fidelity to generate a candidate x,
+        then select the lowest fidelity that has a variance exceeding a
+        threshold.
+
+        Args:
+            candidate_count (int): number of candidates to be generated
+
+        Returns:
+            pd.DataFrame: DataFrame with candidates (proposed experiments)
+        """
+        task_feature: TaskInput = self.domain.inputs.get_by_key(self.task_feature_key)  # type: ignore
+        task_feature.allowed = [True] + [False] * (len(task_feature.categories) - 1)
+        x = super()._ask(candidate_count)
```

Comment:
I have a question about this part: it looks like `task_feature.allowed` doesn't have any influence on the `super()._ask`. Same for selecting the fidelity a couple of lines later. What am I missing here?

Reply:
`task_feature.allowed` has two impacts:
1. By setting `allowed = [True, False, False, ...]`, we fix the fidelity to be the target fidelity (thanks to [1]). This means that when we optimize the function by calling `super()._ask()`, we don't allow the underlying surrogate to optimize over the fidelity (since for the first step of the procedure, we want a candidate x for the target fidelity).
2. Any chosen fidelity must have `allowed[fidelity] == True`, thanks to [2]. We therefore need to set all tasks to be allowed, so that the validation in [2] succeeds

[1] https://github.com/experimental-design/bofire/blob/1266d7109c2af7bd6bd01522145661bf23d5cf1d/bofire/data_models/features/categorical.py#L59-L67
[2] https://github.com/experimental-design/bofire/blob/1266d7109c2af7bd6bd01522145661bf23d5cf1d/bofire/data_models/features/categorical.py#L146-L149